### ISSUE TITLE
`tsh proxy ssh` fails for systems in IoT mode (tunneled to proxy)

### ISSUE THREAD

**Issue by mnaser** [gravitational/teleport#14869]
Expected behavior:
Using `tsh proxy ssh` should work for nodes that are connecting to the proxy service instead of the auth service (tunneled mode).

Current behavior:
When on-boarding infrastructure that does not have direct access to the auth proxy and connecting to the proxy node instead (tunnel mode), everything works for the case of using something like `tsh ssh`, however, trying to proxy to those nodes falls over.

```
❯ ssh -vvv -F ssh_config ctl1.foo.bar
OpenSSH_8.6p1, LibreSSL 3.3.6
debug1: Reading configuration data ssh_config
debug1: ssh_config line 2: Applying options for *.foo.bar
debug1: ssh_config line 9: Applying options for *.foo.bar
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: Executing proxy command: exec "/usr/local/bin/tsh" proxy ssh --cluster=yul1 --proxy=teleport.infra.foo.bar mnaser@ctl1.foo.bar:3022
debug1: identity file /Users/mnaser/.tsh/keys/teleport.infra.foo.bar/mnaser@vexxhost.com type 0
debug1: certificate file /Users/mnaser/.tsh/keys/teleport.infra.foo.bar/mnaser@vexxhost.com-ssh/yul1-cert.pub type 4
debug1: Local version string SSH-2.0-OpenSSH_8.6
ERROR: ssh: subsystem request failed

kex_exchange_identification: Connection closed by remote host
Connection closed by UNKNOWN port 65535
```

Sanitized some stuff to replace it with `.foo.bar`, based on my investigation, this ONLY happens for tunnelled nodes.  FTR, those systems are accessible directly from the client, but cannot talk to the auth service, I also noticed that it isn't listening on port 3022 on those nodes.

Bug details:
- Teleport version: v10.0.2
- Recreation steps
- Debug logs

**Comment by webvictim** [comment]
This is intended to work; can you please share logs from your Teleport auth/proxy service at the time when that `ssh: subsystem request failed` error shows? I suspect there's another error being masked here.

**Comment by mnaser** [comment]
hi @webvictim -- I've troubleshooting this with the Enterprise support team and found out that this only works if I use the SSH config and connecting to `nodename.clustername`, but this happens only to the tunneled nodes.

When trying to connect, I get this on auth/proxy:

```
Jul 26 15:42:07 bastion teleport[39986]: 2022-07-26T15:42:07Z WARN [PROXY]     "Subsystem request proxySubsys(cluster=default/yul1, host=<snip>, port=3022) failed: Teleport proxy failed to connect to \"node\" agent \"<snip>:3022\" over direct dial:\n\n  dial tcp <snip>:3022: connect: connection refused\n\nThis usually means that the agent is offline or has disconnected. Check the\nagent logs and, if the issue persists, try restarting it or re-registering it\nwith the cluster.." id:2790 local:100.112.200.84:443 login:mnaser remote:100.99.168.51:51718 teleportUser:mnaser@vexxhost.com regular/sshserver.go:1775
Jul 26 15:42:07 bastion teleport[39986]: 2022-07-26T15:42:07Z ERRO [NODE]      "Teleport proxy failed to connect to \"node\" agent \"<snip>:3022\" over direct dial:\n\n  dial tcp <snip>:3022: connect: connection refused\n\nThis usually means that the agent is offline or has disconnected. Check the\nagent logs and, if the issue persists, try restarting it or re-registering it\nwith the cluster." regular/sshserver.go:1965
```

I think the main issue here is that could it be possible that when the agent starts in tunnelling mode, it doesn't listen on port 3022?

**Comment by mnaser** [comment]
From what I saw the following:

https://github.com/gravitational/teleport/blob/2da838fca16fb207b6dceaff60edc2c48e66dcdf/lib/service/service.go#L2302-L2354

When the service starts up in tunnel mode, it just assumes that it's not reachable at all.  I think the issue is that I'm not connecting to `nodename.clustername` so Teleport doesn't know to use the Tunnel, and in the other cases, it only worked because it was doing a direct connect to port 3022.

**Comment by mnaser** [comment]
For the record, this was mainly an issue for us in Ansible usage, we've refactored our inventory to include something like this:

```
ansible_host: "{{ inventory_hostname | replace('foo.bar', 'clustername') }}"
```

where cluster name is the teleport cluster name, so it ends up with going to `teleport.infra.clustername` and it's happy afterwards.

**Comment by zmb3** [comment]
Fixed in #14771
